### DISCUSSION TITLE
[Bug]: Switch ACL list CHANGES depending on pagination.

### DISCUSSION THREAD

**Discussion by DeanPomphrett** [mbentley/docker-omada-controller#320] [Help] [answered]
### Controller Version

5.9.31

### Describe the Bug

Issue has been raised on TP-Link Community Forums: https://community.tp-link.com/en/business/forum/topic/610776, as it appears software related, but in case it is related to container, mongo corruption etc I'm referring here as well.

PROBLEM: When on the Site Setting > Network Security > ACL > Switch ACL page, the rules listed AND their statuses change depending on the page or pagination settings.

So for example:

At 5-per-page, a total of 16 rules are listed, at 10-per-page that drops to 12. Notably the difference between these lists is 6 not 4 rules, since the last two rules that are moved onto the second page when set to pagination=10 are NOT included in the 16 shown during pagination=5  Equally, when set to 20-per-page I see 1 page containing the same 16 rules as during pagination=5, albeit with their enabled status toggle to a different values!

Corruption of the data continues in numerous various combinations. A newly added rule for instance is displayed at pagination=5 (as rule 17) missing from pagination=10, 20 & 30 reappears at pagination=40 and is culled again at pagination=50 when the list steps down to showing only 13 rules.

Effectively the page would seem to be versioning different configurations we have deployed, and displaying them associated with different pagination settings.

### Expected Behavior

Rules and their settings  stay consistent across different paginations.

### Steps to Reproduce

1. Deploy a series of switch ACL rules > 10
2. Mutate the list; (Add, Delete Modify)
3. Switch paginations

* This has not been tested from scratch, don't know when the problem occurred.  

### How You're Launching the Container

```plain
I used the compose file provided. Network mode changed from host, to membership of an existing macvlan.
```

### Container Logs

```plain
No associated log activity
```

**Comment by mbentley** [comment]
Yikes, that does not sound awesome at all.  I don't have a TP-Link router so I can't confirm the same behavior on my end.  I wonder if there is any chance that the issue is specific to a certain router/firmware combination or something.  That might be good to share with them.

**Reply by DeanPomphrett** [reply, marked as answer]
Thank you, unfortunately over the weekend I was forced to remove the switch from the controller, revert to web panel admin and then disable L3 routing, so that the office can be ready for this morning. A ticket ID has been issued by TPLink support, if anything of note arises  from that process I will post here for others to be aware of.
